include/net/buf.h

```diff
@@ -400,8 +400,8 @@ static inline void net_buf_simple_restore(struct net_buf_simple *buf,
   */
 struct net_buf {
 	union {
-		/** FIFO uses first 4 bytes itself, reserve space */
```

Comment:
Why was this comment removed? net_buf's still can be passed via FIFO, right? At least the comment should have been left (reworded), yet better to add anon union to make the fact explicit that both fifo and slist are supported.

Reply:
The k_fifo object uses internally sys_slist_t

Reply:
To elaborate a bit more: the k_fifo API doesn't provide any explicit type that should be used to reserve FIFO-internal context. It instead just says "reserve 4 bytes". The 4 byte requirement in turn comes from the fact that k_fifo maps to k_queue which uses sys_slist_t, so having a sys_snode_t in net_buf creates a much more intuitive mapping to k_fifo. That said, having a code comment to explain this would certainly not hurt.

Reply:
@jhedberg : Ack, thanks, makes sense. I'd say, under such conditions, removing the old comment and adding new isn't "incorrect", though if you'd want to elaborate it and still mention fifo, that indeed won't hurt ;-).